### PR TITLE
【6X】NEW SYNTAX for resource group cpuset

### PR DESCRIPTION
**Pull request by CharlieTTXX**
New SYNTAX of resource group cpuset for different master and segment
Using syntax like cpuset="1;3-4" could different cpuset of master and segment by semicolon.  For example,
if we define cpuset="1;3-4", then master will use the first  cpu core, segments will use third  and fourth core.

CREATE and ALTER resource group support this new syntax.
```
postgres=# CREATE RESOURCE GROUP rg1 WITH (cpuset='1;2-3');
CREATE RESOURCE GROUP
postgres=# select groupname, cpuset from gp_toolkit.gp_resgroup_config where groupname = 'rg1';
 groupname | cpuset 
-----------+--------
 rg1       | 1;2-3
(1 row)

postgres=# ALTER resource group rg1 SET cpuset '4;5-6';
ALTER RESOURCE GROUP
postgres=# select groupname, cpuset from gp_toolkit.gp_resgroup_config where groupname = 'rg1';
 groupname | cpuset 
-----------+--------
 rg1       | 4;5-6
(1 row)
```

we didn't make too much motification on Resource group. For differentiate on mater and segment, seperating cpuset by semicolon, then apply the first half of it to master and second half to segment.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Comment by CharlieTTXX** [comment]
7 main branch: https://github.com/greenplum-db/gpdb/pull/14332

File: src/backend/commands/resgroupcmds.c [hunk 1]
```diff
@@ -1030,8 +1032,8 @@ parseStmtOptions(CreateResourceGroupStmt *stmt, ResGroupCaps *caps)
 		if (type == RESGROUP_LIMIT_TYPE_CPUSET) 
 		{
 			const char *cpuset = defGetString(defel);
-			checkCpusetSyntax(cpuset);
 			StrNCpy(caps->cpuset, cpuset, sizeof(caps->cpuset));
```
Comment (kainwen) [inline src/backend/commands/resgroupcmds.c:1035]: Not the fault of this PR. But  `StrNCpy(caps->cpuset, cpuset, sizeof(caps->cpuset));` is not correct. Can we at least search all this kind of bug in resgroup's code and correct them or in another commit?

File: src/backend/commands/resgroupcmds.c [hunk 2]
```diff
@@ -1312,18 +1314,20 @@ validateCapabilities(Relation rel,
 		gp_resource_group_enable_cgroup_cpuset)
 	{
 		Bitmapset *bmsAll = NULL;
+		Bitmapset *bmsMissing = NULL;
```
Comment (kainwen) [inline src/backend/commands/resgroupcmds.c:1317]: Why move it out?

File: src/backend/utils/resgroup/resgroup.c [hunk 3]
```diff
@@ -612,8 +612,9 @@ InitResGroups(void)
 	{
 		ResGroupData	*group;
 		int cpuRateLimit;
-		Oid groupId = HeapTupleGetOid(tuple);
+		Bitmapset *bmsCurrent;
```
Comment (kainwen) [inline src/backend/utils/resgroup/resgroup.c:615]: Why this change?

File: src/backend/commands/resgroupcmds.c (hunk 2 above)
Comment (CharlieTTXX) [reply src/backend/commands/resgroupcmds.c]: done, back it to old place.

File: src/backend/utils/resgroup/resgroup.c (hunk 3 above)
Comment (CharlieTTXX) [reply src/backend/utils/resgroup/resgroup.c]: done, back it to old place.

File: src/backend/commands/resgroupcmds.c (hunk 1 above)
Comment (CharlieTTXX) [reply src/backend/commands/resgroupcmds.c:1035]: Seems `StrNCpy(caps->cpuset, cpuset, strlen(cpuset)); ` pretty make sense.

Comment (kainwen) [reply src/backend/commands/resgroupcmds.c:1035]: I don't think we need copy it. Idea?

Comment (kainwen) [reply src/backend/commands/resgroupcmds.c:1035]: I made a mistake. Sorry.

**Review by kainwen — APPROVED** [review]

**Review by kainwen — APPROVED** [review]
